pkg/flowaggregator/flowaggregator.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flowaggregator
+
+import (
+	"net"
+	"time"
+
+	"github.com/vmware/go-ipfix/pkg/collector"
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/exporter"
+	"github.com/vmware/go-ipfix/pkg/registry"
+	"k8s.io/klog"
+)
+
+type flowAggregator struct {
+	flowCollectorAddr net.Addr
+	exportFrequency   time.Duration
+}
+
+func InitFlowAggregator(flowCollectorAddr net.Addr, exportFrequency time.Duration) *flowAggregator {
+	registry.LoadRegistry()
+	return &flowAggregator{
+		flowCollectorAddr,
+		exportFrequency,
+	}
+}
+
+func (fae *flowAggregator) Run(stopCh <-chan struct{}) {
+	// Go-ipfix TCP collector doesn't work during my last test, fixed as udp temporarily.
+	collectAddr, _ := net.ResolveUDPAddr("udp", "0.0.0.0:4739")
+	cp, _ := collector.InitCollectingProcess(collectAddr, 1024, 0)
+
+	ep, err := exporter.InitExportingProcess(fae.flowCollectorAddr, 1, 0)
+	if err != nil {
+		klog.Fatalf("Got error when connecting to %s", fae.flowCollectorAddr.String())
+	}
+
+	exportTicker := time.NewTicker(fae.exportFrequency)
+	defer exportTicker.Stop()
+	sentMsgNum := 0
+	go cp.Start()
+	for {
+		select {
+		case <-stopCh:
+			break
+		case <-exportTicker.C:
+			curMsgs := cp.GetMessages()
```

Comment:
could you move to go-ipfix v0.3.0 and use the message channel in collector instead?

Reply:
Sure, this PR is based on a commit of go-ipfix before intermediate process merged. In the second PR which I'm working on now, I have moved to go-ipfix v0.3.0 and some other comments will be addressed in that PR too.

Reply:
Ok. Will wait for other PR, but please try not to make PRs bulky. It may be hard for code-reviewers to review.